Correct information about plugin enabling (#1760)
ansible/ansible-documentation: docs/docsite/rst/dev_guide/developing_locally.rst

   Roles contained in collections cannot contain any plugins. All plugins in a collection must live in the collection ``plugins`` directory tree. All plugins in that tree are accessible to all roles in the collection. If you are developing new plugins, we recommend distributing them in :ref:`collections <developing_collections>`, not in roles.

.. warning::

>     Some plugin types are needed early during Ansible execution, like callbacks, inventory and cache, so they cannot be loaded dynamically and are required to exist in configured paths or be referenced via fqcn in configuration.

Reviewer: ```suggestion
    Some plugin types are needed early during Ansible execution, such as callbacks, inventory, and cache. These plugin types cannot be loaded dynamically and must exist in configured paths or be referenced by FQCN in configuration.
```
